File: leetcode/PascalsTriangle/generate.java
public class Solution {
    public ArrayList<ArrayList<Integer>> generate(int numRows) {
        // Start typing your Java solution below
        // DO NOT write main() function
        ArrayList<ArrayList<Integer>> ans = new ArrayList<ArrayList<Integer>>(numRows);
        ArrayList<Integer> row = new ArrayList<Integer>(numRows);
        for (int i = 0; i < numRows; i++) {
            for (int j = i; j >= 0; j--) {
                if (j == i) {
                    int x = i == 0 ? 1 : row.get(i - 1);
                    row.add(x);
                } else {
                    row.set(j, (j == 0 ? 0 : row.get(j - 1)) + row.get(j));
                }
            }
            ans.add(new ArrayList<Integer>(row));
        }
        return ans;
    }
}

File: leetcode/PascalsTriangle/generate.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> ans;
        vector<int> row;
        for (int i = 0; i < numRows; i++) {
            row.insert(row.begin(), 1);
            for (int j = 1; j < i; j++) {
                row[j] = row[j] + row[j + 1];
            }
            ans.push_back(row);
        }
        return ans;
    }
};
